WordPress/Classes/Uploads/UploadsManager.swift

```diff
@@ -0,0 +1,44 @@
+import Foundation
+import Reachability
+
+/// Takes care of coordinating all uploaders used by the app.
+///
+class UploadsManager: NSObject {
+    private let uploaders: [Uploader]
+    private let reachability: Reachability = Reachability.forInternetConnection()
+
+    // MARK: Initialization & Finalization
+
+    /// Default initializer.
+    ///
+    /// - Parameters:
+    ///     - uploaders: the uploaders that this object will be handling.
+    ///
+    required init(uploaders: [Uploader]) {
+        self.uploaders = uploaders
+
+        super.init()
+
+        setupReachableBlock()
+    }
+
+    // MARK: Reachability
+
+    private func setupReachableBlock() {
+        reachability.reachableBlock = { _ in
+            self.resume()
+        }
+
+        reachability.startNotifier()
+    }
+
+    // MARK: Interacting with Uploads
+
+    /// Resumes all uploads handled by the uploaders.
+    ///
+    func resume() {
+        for uploader in uploaders {
```

Comment:
this can also be `uploaders.forEach { $0.resume() }`

Reply:
Changed.